content/v2.0/router/basic-usage.md

```diff
@@ -0,0 +1,267 @@
+---
+title: "Router: Basic Usage"
+order: 20
+---
+
+So far we've only created a very basic endpoint, but usually, our applications will need much more logic to render the page's content.
+
+## Rack
+
+Hanami is compatible with [Rack SPEC](http://www.rubydoc.info/github/rack/rack/master/file/SPEC), and so the endpoints that we use MUST be compliant as well.
+In the example [from the previous section](/v2.0/router/overview) we used a `Proc` that was fitting our requirements.
+
+A valid endpoint can be an object, a class, an action, or an **application** that responds to `#call` and returns the proper value.
+
+```ruby
+# /config/routes.rb
+
+Hanami.application.routes do
+  get "/proc",       to: ->(env) { [200, {}, ["Hello from Hanami!"]] }
+  get "/action",     to: "home.index"
+  get "/middleware", to: Middleware
+  get "/rack-app",   to: RackApp.new
+  get "/rails",      to: ActionControllerSubclass.action(:new)
+end
+```
+
+When we use a string, it tries to instantiate a class from it:
+
+```ruby
+# /config/routes.rb
+
+Hanami.application.routes do
+  get "/rack-app", to: "rack_app" # it will map to RackApp.new
+end
+```
+
+## Actions
+
+Full Rack integration is great, but the most common endpoint that we'll use in our web applications is an **action**.
+Actions are objects responsible for responding to incoming HTTP requests.
+They have a nested naming like `Main::Actions::Home::Index`.
+This is a really long name to write, that's why Hanami has a **naming convention** for it: `"home.index"`.
+
+```ruby
+# /config/routes.rb
+
+Hanami.application.routes do
+  slice :main, at: "/" do
+  root to: "home.index" # => will route to Web::Controllers::Home::Index
+end
+end
+```
+
+The first token is the name of the controller `"home"` is translated to `Home`.
+The same transformation will be applied to the name after the `.`: `"index"` to `Index`.
+
+Hanami can figure out the namespace (`Main::Actions`) based on the slice the route belongs to, and compose the full class name.
+
+```ruby
+# /slices/main/lib/main/actions/home/index.rb
+
+module Main
+  module Actions
+    module Home
+      class Index < Hanami::Action
+      end
+    end
+  end
+end
+```
+
+`Home::Index` action of the `Main` slice, will call a corresponding view object with prepared parameters. The view then will render the prepared `slim` template.
```

Comment:
I can understand including _some_ info about Actions in the Router documentation, since they integrate with each other and it might not be obvious how that happens. I don't think reaching to the next level and talking about views in the router documentation is a good idea though.